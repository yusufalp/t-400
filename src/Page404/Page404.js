import React from 'react';

export default function Page404() {
  return (
    <div className='title not-found'>
      <h2>Page not found!</h2>
      <img src='https://thumbs.gfycat.com/CarefreeWhoppingFluke-max-1mb.gif' alt='hasta la vista, baby' />
    </div>
  )
}
