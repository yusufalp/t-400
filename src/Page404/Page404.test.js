import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Page404 from './Page404';
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

Enzyme.configure({ adapter: new Adapter() });

describe('Page404 component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Page404 />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const tree = renderer
      .create(<Page404 />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders not found message", () => {
    const wrapper = shallow(<Page404 />);
    const notFound = <h2>Page not found!</h2>;
    expect(wrapper.contains(notFound)).toEqual(true);
  });

  it('renders one h2 element that has the message', () => {
    const wrapper = shallow(<Page404 />)
    const message = wrapper.find('h2')
    expect(message).toHaveLength(1)
  });

  it('renders one image', () => {
    const wrapper = shallow(<Page404 />)
    const image = wrapper.find('img')
    expect(image).toHaveLength(1)
  })
})