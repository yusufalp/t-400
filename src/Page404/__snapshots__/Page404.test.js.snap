// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page404 component renders the UI as expected 1`] = `
<div
  className="title not-found"
>
  <h2>
    Page not found!
  </h2>
  <img
    alt="hasta la vista, baby"
    src="https://thumbs.gfycat.com/CarefreeWhoppingFluke-max-1mb.gif"
  />
</div>
`;
